tests/Database/DatabaseEloquentIntegrationTest.php

```diff
@@ -473,6 +501,41 @@ public function testNestedTransactions()
         });
     }
 
+    public function testNestedTransactionsUsingSaveOrFailWillSucceed()
+    {
+        $user = EloquentTestUser::create(['email' => 'taylor@laravel.com']);
+        $this->connection()->transaction(function () use ($user) {
+            try {
+                $user->email = 'otwell@laravel.com';
+                $user->saveOrFail();
+            } catch (Exception $e) {
+                // ignore the exception
+            }
+
+            $user = EloquentTestUser::first();
+            $this->assertEquals('otwell@laravel.com', $user->email);
+            $this->assertEquals(1, $user->id);
+        });
```

Comment:
This tests follow `testNestedTransactions()` with the only different with using `saveOrFail()`.